pulsar-rpc-contrib/src/main/java/org/apache/pulsar/rpc/contrib/client/PulsarRpcClient.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.pulsar.rpc.contrib.client;
+
+import static lombok.AccessLevel.PACKAGE;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+import org.apache.pulsar.rpc.contrib.common.MessageDispatcherFactory;
+
+/**
+ * Provides the functionality to send asynchronous requests and handle replies using Apache Pulsar as the
+ * messaging system. This client manages request-response interactions ensuring that messages are sent
+ * to the correct topics and handling responses through callbacks.
+ *
+ * @param <T> The type of the request messages.
+ * @param <V> The type of the reply messages.
+ */
+@RequiredArgsConstructor(access = PACKAGE)
+public class PulsarRpcClient<T, V> implements AutoCloseable {
```

Comment:
It is recommended to expose the interface externally, refer to PulsarClient.

Reply:
`PulsarRpcClient`, `PulsarRpcClientImpl`, `PulsarRpcClientBuilder`, `PulsarRpcClientBuilderImpl`
Already added. 

PTAL!